inc/policyremoveapplication.class.php

```diff
@@ -90,13 +87,91 @@ public function unicityCheck($value, $itemtype, $itemId, PluginFlyvemdmFleet $fl
     * @return array|bool
     */
    public function getMqttMessage($value, $itemtype, $itemId) {
-      if (! $this->integrityCheck($value, $itemtype, $itemId)) {
+      if (!$this->integrityCheck($value, $itemtype, $itemId)) {
          return false;
       }
 
+      // force the itemtype for packages
+      if (!($itemtype instanceof PluginFlyvemdmPackage)) {
+         $itemtype = new PluginFlyvemdmPackage();
+      }
```

Comment:
$itemtype is a string, not an instance

Also, if the caller sends an unexpected itemtype, it is better to give up with a error code or exception rather than "guessing" what was wanted.

Reply:
That's why I force it, I don't care the value of the itemtype this policies works only for Packages